independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorBindingRegistrar.java

```diff
@@ -1,15 +1,92 @@
 package io.quarkus.arc.processor;
 
+import java.lang.annotation.Annotation;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
 import java.util.Set;
+import java.util.function.Predicate;
 import org.jboss.jandex.DotName;
 
 public interface InterceptorBindingRegistrar extends BuildExtension {
 
     /**
      * Annotations in a form of {@link DotName} to be considered interceptor bindings.
      * Optionally, mapped to a {@link Collection} of non-binding fields
+     * 
+     * @deprecated Use {@link #getAdditionalBindings()} instead.
      */
-    Map<DotName, Set<String>> registerAdditionalBindings();
+    @Deprecated(forRemoval = true)
+    default Map<DotName, Set<String>> registerAdditionalBindings() {
+        return Collections.emptyMap();
+    }
+
+    /**
+     * @return the list of additional interceptor bindings
+     */
+    default List<InterceptorBinding> getAdditionalBindings() {
+        Map<DotName, Set<String>> map = registerAdditionalBindings();
+        if (map.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<InterceptorBinding> bindings = new ArrayList<>();
+        for (Entry<DotName, Set<String>> e : map.entrySet()) {
+            bindings.add(InterceptorBinding.of(e.getKey(), e.getValue()));
+        }
+        return bindings;
+    }
+
+    /**
+     * Represents an additional interceptor binding definition. The name is used to identify an annotation class and the
+     * {@link #test(String)} method is used to match annotation values that should be considered non-binding as defined by the
+     * CDI spec.
+     */
+    interface InterceptorBinding extends Predicate<String> {
+
+        static InterceptorBinding of(Class<? extends Annotation> clazz) {
+            return of(DotName.createSimple(clazz.getName()));
+        }
+
+        static InterceptorBinding of(DotName name) {
+            return of(name, member -> false);
+        }
+
+        static InterceptorBinding of(Class<? extends Annotation> clazz, Predicate<String> predicate) {
```

Comment:
What are the use cases where we need a predicate on the name of the field (in other words, a set of fields is not enough)?

Also, it feels a bit weird that `InterceptorBinding` itself is a `Predicate`.

Other than that, I really like the change, InterceptorBinding wrapper is more readable than the Map

Reply:
Typically all/none of the annotation fields. Imagine an annotation with 10 fields or so and you need to manually list each of them. Not to mention that you'll need to sync if the annotation is changed...